### PR TITLE
feat(impl):[TRI-1063] impl discovery service

### PR DESCRIPTION
**Pull request by ds-ext-kmassalski**
Discovery service rest impl + mock impl.
Returning EDC base url or default address if not found

### CODE REVIEW

File: ess/src/main/java/org/eclipse/tractusx/ess/discovery/EdcAddressResponse.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+/**
+ *
+ */
+@Value
+@Builder
+@Jacksonized
+class EdcAddressResponse {
+
+    private String bpn;
+    private List<String> connectorEndpoint;
```
Comment (ds-jkreutzfeld) [inline ess/src/main/java/org/eclipse/tractusx/ess/discovery/EdcAddressResponse.java:39]: Why should there be multiple endpoints for one BPN?

File: ess/src/main/java/org/eclipse/tractusx/ess/discovery/EdcDiscoveryClient.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+@Service
+class EdcDiscoveryClientLocalStub implements EdcDiscoveryClient {
+
+    @Override
+    public EdcAddressResponse getEdcBaseUrl(final String bpn) {
+        return EdcAddressResponse.builder()
+                                 .bpn(bpn)
+                                 .connectorEndpoint(Collections.singletonList("http://edc-address.com"))
+                                 .build();
+    }
```
[the remark this answers is not in the document]
Comment (ds-jkreutzfeld) [reply ess/src/main/java/org/eclipse/tractusx/ess/discovery/EdcDiscoveryClient.java:58]: The default should be configurable as well.

File: irs-api/src/main/java/org/eclipse/tractusx/irs/configuration/RestTemplateConfig.java [hunk 3]
```diff
@@ -102,6 +103,14 @@ private RestTemplate oAuthRestTemplate(final RestTemplateBuilder restTemplateBui
         return oAuthRestTemplate(restTemplateBuilder, readTimeout, connectTimeout, clientRegistrationId);
     }
 
+    @Bean(DISCOVERY_REST_TEMPLATE)
+        /* package */ RestTemplate discoveryRestTemplate(final RestTemplateBuilder restTemplateBuilder,
+            @Value("${bpdm.timeout.read}") final Duration readTimeout,
+            @Value("${bpdm.timeout.connect}") final Duration connectTimeout,
+            @Value("${bpdm.oAuthClientId}") final String clientRegistrationId) {
```
Comment (ds-jkreutzfeld) [inline irs-api/src/main/java/org/eclipse/tractusx/irs/configuration/RestTemplateConfig.java:110]: These entries are missing in the config, right?

File: ess/src/main/java/org/eclipse/tractusx/ess/discovery/EdcAddressResponse.java (hunk 1 above)
Comment (ds-ext-kmassalski) [reply ess/src/main/java/org/eclipse/tractusx/ess/discovery/EdcAddressResponse.java:39]: Based on documentation: https://confluence.catena-x.net/display/CORE/Dataspace+Discovery

File: ess/src/main/java/org/eclipse/tractusx/ess/discovery/EdcDiscoveryFacade.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * Public API Facade for Discovery service domain
+ */
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class EdcDiscoveryFacade {
+
+    private static final String DEFAULT_EDC_ENDPOINT = "http://default";
```
[the remark this answers is not in the document]
Comment (ds-ext-kmassalski) [reply ess/src/main/java/org/eclipse/tractusx/ess/discovery/EdcDiscoveryFacade.java]: We agreed to not use any defaults, but return "Unknown" status

File: irs-api/src/main/java/org/eclipse/tractusx/irs/configuration/RestTemplateConfig.java (hunk 3 above)
Comment (ds-ext-kmassalski) [reply irs-api/src/main/java/org/eclipse/tractusx/irs/configuration/RestTemplateConfig.java]: Done, without charts. Is it enough or add it also there?

**Review by ds-jkreutzfeld — CHANGES_REQUESTED** [review]

File: irs-api/src/main/java/org/eclipse/tractusx/irs/configuration/RestTemplateConfig.java (hunk 3 above)
Comment (ds-jkreutzfeld) [reply irs-api/src/main/java/org/eclipse/tractusx/irs/configuration/RestTemplateConfig.java]: That's okay for now.

**Comment by ds-jhartmann** [comment]
Merged by accident, wanted to merge my own PR, sorry!

**Pull request merged**